### PR TITLE
test: Create spore integration tests

### PR DESCRIPTION
**Pull request by Dawn-githup**
## Changes
- Created spore integration tests.
- Refactored workflow to enable parallel testing, improving test efficiency.
- Renamed log files for better clarity and readability.

### CODE REVIEW

**Review by duanyytop — DISMISSED** [review]

File: .github/workflows/integration-test.yaml [hunk 1]
```diff
@@ -33,9 +33,11 @@ jobs:
       - name: Build packages
         run: pnpm run build:packages
 
-      - name: Run integration:xudt script
+      - name: Run integration:xudt/spore script
         working-directory: ./tests/rgbpp
-        run: pnpm run integration:xudt
+        run: |
+          pnpm run integration:xudt
+          pnpm run integration:spore
```
Comment (Flouse) [inline .github/workflows/integration-test.yaml:40]: If `integration:xudt` and `integration:spore` use different keys, they can run in prarallel, right?

**motivation**: To optimize the workflow and reduce the time consumed.

### Reference
- https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix

File: tests/rgbpp/spore/launch/1-prepare-cluster.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    console.info(`Cluster cell has been prepared and the tx hash ${txHash}`);
+    console.info(`explorer: https://pudge.explorer.nervos.org/transaction/${txHash}`);
+  });
+};
+
+// Please use your real BTC UTXO information on the BTC Testnet
+prepareClusterCell({
+  // outIndex: 3,
+  // btcTxId: 'aee4e8e3aa95e9e9ab1f0520714031d92d3263262099dcc7f7d64e62fa2fcb44',
+  outIndex: readStepLog('0').index,
```
Comment (Flouse) [inline tests/rgbpp/spore/launch/1-prepare-cluster.ts:82]: The Spore log files are indistinguishable from xUDT log files.

File: tests/rgbpp/spore/launch/2-create-cluster.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    const feeRate = await getFastestFeeRate();
+    console.log('feeRate = ', feeRate);
+
+    const ckbVirtualTxResult = await genCreateClusterCkbVirtualTx({
+      collector,
+      rgbppLockArgs: ownerRgbppLockArgs,
+      clusterData: CLUSTER_DATA,
+      isMainnet,
+      ckbFeeRate: BigInt(2000),
+    });
```
Comment (Flouse) [inline tests/rgbpp/spore/launch/2-create-cluster.ts:28]: ### suggestion
Also follow the best practice -> `saveCkbVirtualTxResult` in the tests.

- ref: https://github.com/ckb-cell/rgbpp-sdk/issues/192

Comment (Flouse) [inline tests/rgbpp/spore/launch/2-create-cluster.ts:28]: ### suggestion
Also follow the best practice -> `saveCkbVirtualTxResult` in the tests.

- ref: https://github.com/ckb-cell/rgbpp-sdk/issues/192

File: tests/rgbpp/spore/launch/3-create-spores.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    const feeRate = await getFastestFeeRate();
+    console.log('feeRate = ', feeRate);
+
+    const ckbVirtualTxResult = await genCreateSporeCkbVirtualTx({
+      collector,
+      sporeDataList: receivers.map((receiver) => receiver.sporeData),
+      clusterRgbppLockArgs,
+      isMainnet,
+      ckbFeeRate: BigInt(2000),
+    });
```
Comment (Flouse) [inline tests/rgbpp/spore/launch/3-create-spores.ts:46]: ### suggestion
Also follow the best practice -> `saveCkbVirtualTxResult` in the tests.

- ref: https://github.com/ckb-cell/rgbpp-sdk/issues/192

File: tests/rgbpp/spore/launch/1-prepare-cluster.ts (hunk 2 above)
Comment (Dawn-githup) [reply tests/rgbpp/spore/launch/1-prepare-cluster.ts:82]: Yes, a tx_id coverage log file will be re-created before the spore test starts. If it is modified to a parallel test, it needs to be renamed.

File: tests/rgbpp/spore/4-transfer-spore.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    const feeRate = await getFastestFeeRate();
+    console.log('feeRate = ', feeRate);
+
+    const ckbVirtualTxResult = await genTransferSporeCkbVirtualTx({
+      collector,
+      sporeRgbppLockArgs,
+      sporeTypeBytes,
+      isMainnet,
+    });
```
Comment (Flouse) [inline tests/rgbpp/spore/4-transfer-spore.ts:30]: ditto

File: tests/rgbpp/spore/5-leap-spore-to-ckb.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    const feeRate = await getFastestFeeRate();
+    console.log('feeRate = ', feeRate);
+
+    const ckbVirtualTxResult = await genLeapSporeFromBtcToCkbVirtualTx({
+      collector,
+      sporeRgbppLockArgs,
+      sporeTypeBytes,
+      toCkbAddress,
+      isMainnet,
+    });
```
Comment (Flouse) [inline tests/rgbpp/spore/5-leap-spore-to-ckb.ts:31]: ditto

File: .github/workflows/integration-test.yaml [hunk 7]
```diff
@@ -33,12 +38,25 @@ jobs:
...
           VITE_SERVICE_URL: ${{ secrets.SERVICE_URL }}
           VITE_SERVICE_TOKEN: ${{ secrets.SERVICE_TOKEN }}
           VITE_SERVICE_ORIGIN: ${{ secrets.SERVICE_ORIGIN }}
           INTEGRATION_CKB_PRIVATE_KEY: ${{ secrets.INTEGRATION_CKB_PRIVATE_KEY }}
           INTEGRATION_BTC_PRIVATE_KEY: ${{ secrets.INTEGRATION_BTC_PRIVATE_KEY }}
+
+      - name: Run integration:spore script
+        working-directory: ./tests/rgbpp
+        if: ${{ matrix.env_set == 'spore' }}
+        run: pnpm run integration:spore
```
Comment (Flouse) [inline .github/workflows/integration-test.yaml:56]: Has `pnpm run integration:spore` been tested anywhere?

File: tests/rgbpp/spore/launch/2-create-cluster.ts [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+    // Save ckbVirtualTxResult
+    saveCkbVirtualTxResult(ckbVirtualTxResult, '2-create-cluster');
+
+    const { commitment, ckbRawTx, clusterId, needPaymasterCell } = ckbVirtualTxResult;
+
+    writeStepLog('3', {
```
Comment (Flouse) [inline tests/rgbpp/spore/launch/2-create-cluster.ts:36]: ### Suggestion
Using clear step names is better.

File: tests/rgbpp/spore/launch/3-create-spores.ts [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+      feeRate: feeRate,
+    });
+    psbt.signAllInputs(btcKeyPair);
+    psbt.finalizeAllInputs();
+
+    const btcTx = psbt.extractTransaction();
+    const btcTxBytes = transactionToHex(btcTx, false);
+    const { txid: btcTxId } = await btcService.sendBtcTransaction(btcTx.toHex());
+
+    writeStepLog('5', {
```
Comment (Flouse) [inline tests/rgbpp/spore/launch/3-create-spores.ts:74]: ### Suggestion
Using clear step names is better.

File: tests/rgbpp/spore/launch/3-create-spores.ts [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+          ckbRawTx: newCkbRawTx,
+          btcTxBytes,
+          rgbppApiSpvProof,
+        });
+
+        // The outputs[1..] are spore cells from which you can find spore type scripts,
+        // and the spore type scripts will be used to transfer and leap spores
+        const sporeTypeScripts = ckbTx.outputs.slice(1).map((output) => output.type);
+        console.log('Spore type scripts: ', JSON.stringify(sporeTypeScripts));
+        writeStepLog('6', sporeTypeScripts);
```
Comment (Flouse) [inline tests/rgbpp/spore/launch/3-create-spores.ts:99]: ### Suggestion
Using clear step names is better.

**Review by Flouse — DISMISSED** [review]

File: .github/workflows/integration-test.yaml (hunk 7 above)
Comment (Dawn-githup) [reply .github/workflows/integration-test.yaml:56]: https://github.com/ckb-cell/rgbpp-sdk/actions/runs/9405737494/job/25907598618?pr=210

**Review by Flouse — APPROVED** [review]

**Comment by Flouse** [comment]
successful run: https://github.com/ckb-cell/rgbpp-sdk/actions/runs/9405737494

**Pull request merged**